libs/libcommon/include/common/find_first_symbols.h

```diff
@@ -83,7 +87,7 @@ template <size_t num_chars,
 	char c13 = 0, char c14 = 0, char c15 = 0, char c16 = 0>
 inline const char * find_first_symbols_sse42_impl(const char * begin, const char * end)
 {
-#if defined(__x86_64__)
+#if HAVE_SSE42
```

Comment:
Need to move this check out of the function.
Otherwise you will get poor performance when SSE 4.2 is not available, but SSE 2 is.

Also note, that x86_64 always have SSE 2.